diff --git a/cli/lmi/shell/LMIIndicationListener.py b/cli/lmi/shell/LMIIndicationListener.py
index 1ec2f21..cc48b9b 100644
--- a/cli/lmi/shell/LMIIndicationListener.py
+++ b/cli/lmi/shell/LMIIndicationListener.py
@@ -115,6 +115,9 @@ class LMIIndicationListener(object):
         self._certfile = certfile
         self._keyfile = keyfile
 
+        self._server = None
+        self._server_thread = None
+
     def __create_handler_name(self, handler_name_pattern):
         """
         Returns unique handler name by replacing "**X**" characters for random characters
@@ -163,8 +166,12 @@ class LMIIndicationListener(object):
         """
         Stops the indication listener.
         """
-        self._server.shutdown()
-        self._server_thread.join()
+        if self._server:
+            self._server.shutdown()
+        if self._server_thread:
+            self._server_thread.join()
+        self._server = None
+        self._server_thread = None
 
     @property
     def is_alive(self):
@@ -172,7 +179,9 @@ class LMIIndicationListener(object):
         :returns: flag indicating, if the indication listener is running
         :rtype: bool
         """
-        return self._server_thread.is_alive()
+        if self._server_thread:
+            return self._server_thread.is_alive()
+        return False
 
     @property
     def hostname(self):
